test/Microsoft.AspNetCore.OData.Tests/Query/Expressions/FilterBinderTests.cs

```
@@ -2639,38 +2642,59 @@ public void IsOfQuotedUndefinedTarget_ThrowsODataException(string filter)
     }
 
     [Theory]
-    [InlineData("isof(Microsoft.AspNetCore.OData.Tests.Models.Address)")]
-    [InlineData("isof(null,Microsoft.AspNetCore.OData.Tests.Models.Address)")]
-    [InlineData("isof(null, Microsoft.AspNetCore.OData.Tests.Models.Address)")]
-    [InlineData("isof(SupplierAddress,Microsoft.AspNetCore.OData.Tests.Models.Address)")]
-    [InlineData("isof(SupplierAddress, Microsoft.AspNetCore.OData.Tests.Models.Address)")]
-    public void IsOfUnquotedComplexType_ThrowsODataException(string filter)
+    [InlineData("isof(Microsoft.AspNetCore.OData.Tests.Models.Address)", "Microsoft.AspNetCore.OData.Tests.Models.Product")]
+    [InlineData("isof(null,Microsoft.AspNetCore.OData.Tests.Models.Address)", "<null>")]
+    [InlineData("isof(null, Microsoft.AspNetCore.OData.Tests.Models.Address)", "<null>")]
+    public void IsOfUnquotedComplexType_ThrowsNotAssignableFromException(string filter, string assignableFrom)
     {
         // Arrange
         var expectedMessage =
             "Encountered invalid type cast. " +
-            "'Microsoft.AspNetCore.OData.Tests.Models.Address' is not assignable from 'Microsoft.AspNetCore.OData.Tests.Models.Product'.";
+            $"'Microsoft.AspNetCore.OData.Tests.Models.Address' is not assignable from '{assignableFrom}'.";
```

I wonder why `isof` throws an exception when the cast is not possible instead of returning `false`. Is that how it has always worked? It seems counter-intuitive to me.

---

I will create a different PR with `isof` and `cast` support